### PR TITLE
Suppress server log message when EOF without receiving data for preface

### PR DESCRIPTION
**Pull request by dbudworth**
When using a TCP load balancer (ie: AWS ELB), they do health checks by simply opening a connection to your port and then closing it.  

Just to test if you can accept a connection.

Currently, this causes a log message of:
```
transport: http2Server.HandleStreams failed to receive the preface from client: EOF
```
This change suppresses that log message if, and only if, no bytes were received.

Basically, it allows a GRPC server to be used behind a simple TCP load balancer without spamming your logs.

### CODE REVIEW

**Comment by MakMukhi** [comment]
@menghanl  is already working on adding different levels to logging. This should be taken care of there. Menghan can confirm further when he's back.

**Comment by dbudworth** [comment]
@menghanl Any chance your changes are going in soon?  I have to keep moving the patch forward when I pull due to current code being super-noisy when used behind a tcp balancer.

**Comment by menghanl** [comment]
The first step (#922) should be done by the end of this week.
After that, we will need another PR to change the logging levels.

File: transport/http2_server.go [hunk 1]
```diff
@@ -274,8 +274,11 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 func (t *http2Server) HandleStreams(handle func(*Stream), traceCtx func(context.Context, string) context.Context) {
 	// Check the validity of client preface.
 	preface := make([]byte, len(clientPreface))
-	if _, err := io.ReadFull(t.conn, preface); err != nil {
-		grpclog.Printf("transport: http2Server.HandleStreams failed to receive the preface from client: %v", err)
+	if n, err := io.ReadFull(t.conn, preface); err != nil {
+		// Only log if it isn't a simple tcp accept check (ie: tcp balancer doing open/close socket)
+		if n > 0 || err != io.EOF {
```
Comment (menghanl) [inline transport/http2_server.go]: I believe we just need to check `io.EOF` here. If n>0, err will be `ErrUnexpectedEOF`.

From the doc of `ReadFull`:

    The error is EOF only if no bytes were read.
    If an EOF happens after reading some but not all the bytes,
    ReadFull returns ErrUnexpectedEOF.

Comment (dbudworth) [inline transport/http2_server.go]: makes sense, updated...

**Pull request merged**